packages/backend/src/peripherals/database/activity-v2/ZksyncTransactionRepository.ts

```diff
@@ -0,0 +1,44 @@
+import { Logger } from '@l2beat/common'
+import { UnixTime } from '@l2beat/types'
+import { Knex } from 'knex'
+import { ZksyncTransactionRow } from 'knex/types/tables'
+
+import { BaseRepository } from '../shared/BaseRepository'
+import { Database } from '../shared/Database'
+
+export interface ZksyncTransactionRecord {
+  blockNumber: number
+  blockIndex: number
+  timestamp: UnixTime
+}
+
+export class ZksyncTransactionRepository extends BaseRepository {
+  constructor(database: Database, logger: Logger) {
+    super(database, logger)
+
+    /* eslint-disable @typescript-eslint/unbound-method */
+    this.addMany = this.wrapAddMany(this.addMany)
+    this.deleteAll = this.wrapDelete(this.deleteAll)
+    /* eslint-enable @typescript-eslint/unbound-method */
+  }
+
+  async addMany(records: ZksyncTransactionRecord[], trx?: Knex.Transaction) {
+    const knex = await this.knex()
+    const rows = records.map(toRow)
+    await (trx ?? knex)('activity_v2.zksync').insert(rows)
```

Comment:
I propose a different approach, which also allows us to wait for migrations.
```ts
const knex = await this.knex(trx)
```
And then in `Database.getKnex()`
```ts
  async getKnex(trx?: Knex.Transaction) {
    if (!this.migrated) {
      await this.migrationsComplete
    }
    return trx ?? this.knex
  }
```